torch/_tensor.py

```diff
@@ -539,6 +539,56 @@ def register_hook(self, hook):
         self._backward_hooks[handle.id] = hook
         return handle
 
+    def register_post_accumulate_grad_hook(self, hook):
+        r"""Registers a backward hook that runs after grad accumulation.
+
+        The hook will be called after all gradients for a tensor have been accumulated,
+        meaning that the .grad field has been updated on that tensor. The post
+        accumulate grad hook is ONLY applicable for leaf tensors as the AccumulateGrad
+        node is only called for leaf tensors. Registering this hook on a non-leaf
+        tensor will do nothing!
```

Comment:
And so should be an error if there is a grad_fn field!

Reply:
There might be a caveat here for Tensor where retain_grad() is set. But that can be a follow up ;p

Reply:
Hm, it wouldn't work even when retains grad is set..